apps/files/src/components/FileList.vue

```diff
@@ -154,8 +155,23 @@ export default {
       this.$emit('FileAction', file)
     },
     checkNewName (name) {
-      if (/[/]/.test(name)) {
-        return this.$gettext('Name cannot contain "/"')
+      if (/[/]/.test(name)) return this.$gettext('Name cannot contain "/"')
+
+      if (name === '.') return this.$gettext('Name cannot be equal to "."')
+
+      if (name === '..') return this.$gettext('Name cannot be equal to ".."')
+
+      if (/\s+$/.test(name)) return this.$gettext('Name cannot end with whitespace')
+
+      const exists = this.activeFiles.find((n) => {
+        if (n['name'] === name && this.originalName !== name) {
+          return n
+        }
+      })
+
+      if (exists) {
+        const translatedDesc = this.$gettext('%{name} already exists')
```

Comment:
A file with the name %{name} already exists

DO we know if this is a file or a folder?

Reply:
The exists var contains the file so we can see the type. I was just thinking that it could maybe make the user confused when he's renaming a file and see an error message saying "A folder with the name..." and the other way round so I dropped the words file and folder from it. But I can go ahead and implement it there.

Reply:
I changed the message now to something different to make it a little bit more descriptive and still prevent the issue I have with the words folder and file in it. Let me know what do you think about it.